tests/functional/object/mcg/test_cors_config_elements.py

```diff
@@ -246,39 +348,351 @@ def test_basic_cors_operations(
+            cors_config, awscli_pod_session, bucket_name, mcg_obj_session
+        )
+
+        # 5: Perform multiple ops from allowed method request from allowed origin
+        for i in cors_config["CORSRules"][0]["AllowedMethods"]:
+            sample = TimeoutSampler(
+                timeout=120,
+                sleep=10,
+                func=self.exec_curl_command,
+                endpoint=mcg_obj_session.s3_endpoint,
+                bucket_name=bucket_name,
+                origin=allowed_origin,
+                method=i,
+            )
+            sample.wait_for_func_value(self.POSITIVE_RESPONSE)
+
+        # 6: Modify the existing CORS config and add non-supported HTTP method in it(PATCH)
+        cors_config["CORSRules"][0]["AllowedMethods"].append("PATCH")
+        try:
+            resp = self.apply_cors_on_bucket(
+                cors_config,
+                awscli_pod_session,
+                bucket_name,
+                mcg_obj_session,
+                response=True,
+            )
+            assert resp is None, "CORS is applying invalid HTTP method(PATCH)"
+        except Exception as e:
+            if "InvalidRequest" in str(e):
+                logger.warning("Expected error")
+                logger.warning(e)
+            else:
+                raise
+
+        # 7: Modify the existing CORS config and add all suported HTTP method along with multiple origins in it
+        cors_config["CORSRules"][0]["AllowedMethods"] = []
+        for val in self.allowed_methods_list:
+            cors_config["CORSRules"][0]["AllowedMethods"].append(val)
+        cors_config["CORSRules"][0]["AllowedOrigins"] = ["*"]
+        self.apply_cors_on_bucket(
+            cors_config, awscli_pod_session, bucket_name, mcg_obj_session
+        )
+
+        # 8: Perform any request mentioned in allowed HTTP method from any supported origin mentioned on CORS
+        for i in range(3):
+            method = choice(self.allowed_methods_list)
+            origin = self.generate_random_domain_address()[0]
+            sample = TimeoutSampler(
+                timeout=120,
+                sleep=10,
+                func=self.exec_curl_command,
+                endpoint=mcg_obj_session.s3_endpoint,
+                bucket_name=bucket_name,
+                origin=origin,
+                method=method,
+            )
+            sample.wait_for_func_value(self.POSITIVE_RESPONSE)
+
+    @tier2
+    def test_MaxAgeSeconds_and_AllowCredentials_element(
+        self, mcg_obj_session, awscli_pod_session, bucket_factory
+    ):
+        """
+        Test MaxAgeSeconds and AllowCredentials element from CORS operation on bucket
+            step #1: Create bucket and apply basic CORS config with MaxAgeSeconds element in it
+            step #2: Modify MaxAgeSeconds parameter by adding 30 secs in it
+            step #3: Apply basic CORS config on bucket by adding "AllowCredentials" parameter to "False" value
+        """
```

Comment:
```suggestion
        """
        Test MaxAgeSeconds and AllowCredentials element from CORS operation on bucket
            step #1: Create bucket and apply basic CORS config with MaxAgeSeconds element in it
            step #2: Modify MaxAgeSeconds parameter by adding 30 secs in it and validate the same
            step #3: Apply basic CORS config on bucket by adding "AllowCredentials" parameter to "False" value
        """
```

Regarding the third step, it looks like we're expecting a CORS config with this rule to be Rejected (regardless if the value is true or false). From what I've gathered, this is supported in AWS S3 but not in MCG at this point in time where this feature is added. If this is the case, please include a brief sumamry of this background to the docstring in order clarify this expectation from the test.

Reply:
Fixed